INPTP_AppForFixing/EmployeeDialog.cs

```diff
@@ -1,59 +1,92 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.ComponentModel;
-using System.Data;
-using System.Drawing;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Windows.Forms;
 
 namespace INPTP_AppForFixing
 {
     public partial class EmployeeDialog : Form
     {
         private MainForm main;
-        private bool employee=false;
+        private bool employee = false;
+        private Boss bossForEmpl;
 
-        public EmployeeDialog(MainForm main, bool employee)
+        public EmployeeDialog(MainForm main, bool employee, Boss selectedBoss = null)
         {
             InitializeComponent();
-            this.main = main;
-            this.employee = employee;
-            if (employee) {
-                lbDep.Visible = false;
-                lbDpName.Visible = false;
-                tBDpName.Visible = false;
+            Init(main, employee);
+            bossForEmpl = selectedBoss;
+            if (selectedBoss != null && employee == false)
+            {
+                tBID.Text = selectedBoss.Id.ToString();
+                tBID.Enabled = false;
+                tBFirstName.Text = selectedBoss.FirstName;
+                tBLastName.Text = selectedBoss.LastName;
+                tBJob.Text = selectedBoss.Job;
+                tBSalary.Text = selectedBoss.MonthlySalaryCZK.ToString();
+                dateTimePickerBirthDate.Value = selectedBoss.OurBirthDate;
             }
         }
 
         private void btnOk_Click(object sender, EventArgs e)
         {
-            if (testOfValues())
+            if (validFormInput())
             {
-                main.Bosses.Add(new Boss(Int32.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text)));
-                main.OnBossesChange();
+                if (employee)
+                {
+                    Employee temp = new Employee(int.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text));
+                    bossForEmpl.InsertEmpl(temp);
+                    main.OnEmplChange();
+                }
+                else
+                {
+                    Boss newBoss = new Boss(int.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text));
+
+                    // Find boss in hashset by boss ID
```

Comment:
comments that tells us what code does are bad comments. Comments should explain why you did this...

Reply:
I understand